libs/index/odc/index/stac.py

```diff
@@ -86,11 +108,16 @@ def stac_transform(input_stac):
     """ Takes in a raw STAC 1.0 dictionary and returns an ODC dictionary
     """
 
-    product_label, product_name, region_code = _stac_product_lookup(input_stac)
+    product_label, product_name, region_code, default_grid = _stac_product_lookup(input_stac)
 
-    deterministic_uuid = str(odc_uuid("sentinel-2_stac_process", "1.0.0", [product_label]))
+    # Generating UUID for products not having UUID.
+    # TODO: Check is based on hardcoded Product Name, find generic way
+    if product_name in ["s2_l2a"]:
+        deterministic_uuid = str(odc_uuid("sentinel-2_stac_process", "1.0.0", [product_label]))
+    else:
+        deterministic_uuid = input_stac["id"]
```

Comment:
is `id` keys guaranteed by STAC api?
what happens if it's not there?

Reply:
The `id` is REQUIRED field and will exists in any valid STAC doc.
https://github.com/radiantearth/stac-spec/blob/master/item-spec/item-spec.md#item-fields

Reply:
ok, but it's not UUID, code elsewhere assumes UUID. Need to generate deterministic UUID somehow.